Description: Do not write desktop entries on starting
 The present patch skips the registration of URL scheme at runtime. An already
 packaged org.telegram.desktop.desktop file provides the needed features.
Discussion: https://github.com/telegramdesktop/tdesktop/pull/8627#issuecomment-720578885
Author: Nicholas Guriev <guriev-ns@ya.ru>
Last-Update: Tue, 03 Oct 2023 08:14:47 +0300

--- a/Telegram/lib_base/CMakeLists.txt
+++ b/Telegram/lib_base/CMakeLists.txt
@@ -54,6 +54,7 @@ PRIVATE
     base/platform/linux/base_process_linux.h
     base/platform/linux/base_system_media_controls_linux.cpp
     base/platform/linux/base_url_scheme_linux.cpp
+    base/platform/linux/base_url_scheme_linux_dummy.cpp
     base/platform/linux/base_url_scheme_linux.h
     base/platform/mac/base_battery_saving_mac.mm
     base/platform/mac/base_battery_saving_mac.h
@@ -239,6 +240,16 @@ else()
     )
 endif()
 
+if (DESKTOP_APP_USE_PACKAGED OR GLIBMM_VERSION VERSION_LESS 2.76)
+    remove_target_sources(lib_base ${src_loc}
+        base/platform/linux/base_url_scheme_linux.cpp
+    )
+else()
+    remove_target_sources(lib_base ${src_loc}
+        base/platform/linux/base_url_scheme_linux_dummy.cpp
+    )
+endif()
+
 target_include_directories(lib_base
 PUBLIC
     ${src_loc}
--- /dev/null
+++ b/Telegram/lib_base/base/platform/linux/base_url_scheme_linux_dummy.cpp
@@ -0,0 +1,13 @@
+// This file is provided in Debian to mask unneeded function.
+//
+// Licence: CC0-1.0, https://creativecommons.org/publicdomain/zero/1.0/legalcode
+//
+#include "base/platform/linux/base_url_scheme_linux.h"
+
+namespace base::Platform {
+
+void RegisterUrlScheme(const UrlSchemeDescriptor &) {
+	// do nothing
+}
+
+} // namespace base::Platform
